x-pack/plugins/cases/public/components/user_actions/index.tsx

```diff
@@ -0,0 +1,269 @@
+          onUpdateField,
+          handleManageMarkdownEditId,
+          handleManageQuote,
+        }),
+      [
+        caseData,
+        commentRefs,
+        manageMarkdownEditIds,
+        isLoadingDescription,
+        userCanCrud,
+        onUpdateField,
+        handleManageMarkdownEditId,
+        handleManageQuote,
+      ]
+    );
+
+    const userActions: EuiCommentProps[] = useMemo(
+      () =>
+        caseUserActions.reduce<EuiCommentProps[]>(
+          (comments, userAction, index) => {
+            if (isUserActionTypeSupported(userAction.type)) {
+              const builder = builderMap[userAction.type];
+              const userActionBuilder = builder({
+                caseData,
+                userAction,
+                caseServices,
+                comments: caseData.comments,
+                index,
+                userCanCrud,
+                commentRefs,
+                manageMarkdownEditIds,
+                selectedOutlineCommentId,
+                loadingCommentIds,
+                loadingAlertData,
+                alertData: manualAlertsData,
+                handleOutlineComment,
+                handleManageMarkdownEditId,
+                handleSaveComment,
+                handleManageQuote,
+                onShowAlertDetails,
+                actionsNavigation,
+                getRuleDetailsHref,
+                onRuleDetailsClick,
+              });
+              return [...comments, ...userActionBuilder.build()];
+            }
+            return comments;
+          },
+          [descriptionCommentListObj]
+        ),
+      [
+        caseUserActions,
+        descriptionCommentListObj,
+        caseData,
+        caseServices,
+        userCanCrud,
+        commentRefs,
+        manageMarkdownEditIds,
+        selectedOutlineCommentId,
+        loadingCommentIds,
+        loadingAlertData,
+        manualAlertsData,
+        handleOutlineComment,
+        handleManageMarkdownEditId,
+        handleSaveComment,
+        handleManageQuote,
+        onShowAlertDetails,
+        actionsNavigation,
+        getRuleDetailsHref,
+        onRuleDetailsClick,
+      ]
+    );
+
+    const bottomActions = userCanCrud
+      ? [
+          {
+            username: (
+              <UserActionUsername
+                username={currentUser?.username}
+                fullName={currentUser?.fullName}
+              />
+            ),
+            'data-test-subj': 'add-comment',
+            timelineIcon: (
+              <UserActionAvatar username={currentUser?.username} fullName={currentUser?.fullName} />
+            ),
+            className: 'isEdit',
+            children: MarkdownNewComment,
+          },
+        ]
+      : [];
+
+    const comments = [...userActions, ...bottomActions];
+
+    return (
+      <>
+        <MyEuiCommentList comments={comments} data-test-subj="user-actions" />
+        {(isLoadingUserActions || loadingCommentIds.includes(NEW_COMMENT_ID)) && (
```

Comment:
could you turn this into a ternary? It makes it easier to read.

Reply:
I like better this pattern to be honest. With a ternary, I have to return `null` and I found it more difficult to read.

Reply:
@cnasikas I understand that but keep in mind it is more common for react to have ternaries returning null than having expressions like this. I  won't push for the change but I find it better to follow code conventions like that one.